include/event_manager/event_manager_priv.h

```diff
@@ -0,0 +1,220 @@
+ * an argument. Allocator function is used to create an event of the given
+ * ename type.
+ */
+#define _EVENT_ALLOCATOR_DYNDATA_FN(ename)						\
+	static inline struct ename *_CONCAT(new_, ename)(size_t size)			\
+	{										\
+		struct ename *event =							\
+			(struct ename *)event_manager_alloc(sizeof(*event) + size);	\
+		BUILD_ASSERT((offsetof(struct ename, dyndata) +				\
+				  sizeof(event->dyndata.size)) ==			\
+				 sizeof(*event), "");					\
+		BUILD_ASSERT(offsetof(struct ename, header) == 0,			\
+				 "");							\
+		event->header.type_id = _EVENT_ID(ename);				\
+		event->dyndata.size = size;						\
+		return event;								\
+	}
+
+
+/* Macro generates a function of name cast_ename where ename is provided as
+ * an argument. Casting function is used to convert event_header pointer
+ * into pointer to event matching the given ename type.
+ */
+#define _EVENT_CASTER_FN(ename)									\
+	static inline struct ename *_CONCAT(cast_, ename)(const struct event_header *eh)	\
+	{											\
+		struct ename *event = NULL;							\
+		if (eh->type_id == _EVENT_ID(ename)) {						\
+			event = CONTAINER_OF(eh, struct ename, header);				\
+		}										\
+		return event;									\
+	}
+
+
+/* Macro generates a function of name is_ename where ename is provided as
+ * an argument. Typecheck function is used to check if pointer to event_header
+ * belongs to the event matching the given ename type.
+ */
+#define _EVENT_TYPECHECK_FN(ename) \
+	static inline bool _CONCAT(is_, ename)(const struct event_header *eh)	\
+	{									\
+		return (eh->type_id == _EVENT_ID(ename));			\
+	}
+
+
+
+/* Declarations and definitions - for more details refer to public API. */
+#define _EVENT_LISTENER(lname, notification_fn)					\
+	const struct event_listener _CONCAT(__event_listener_, lname) __used	\
+	__attribute__((__section__("event_listeners"))) = {			\
+		.name = STRINGIFY(lname),					\
+		.notification = (notification_fn),				\
+	}
+
+
+#define _EVENT_TYPE_DECLARE_COMMON(ename)				\
+	extern const struct event_type _CONCAT(__event_type_, ename);	\
+	_EVENT_SUBSCRIBERS_DECLARE(ename);				\
+	_EVENT_CASTER_FN(ename);					\
+	_EVENT_TYPECHECK_FN(ename)
+
+
+#define _EVENT_TYPE_DECLARE(ename)					\
+	_EVENT_TYPE_DECLARE_COMMON(ename);				\
+	_EVENT_ALLOCATOR_FN(ename)
+
+
+#define _EVENT_TYPE_DYNDATA_DECLARE(ename)				\
+	_EVENT_TYPE_DECLARE_COMMON(ename);				\
+	_EVENT_ALLOCATOR_DYNDATA_FN(ename)
+
+
+#define _EVENT_TYPE_DEFINE(ename, init_log_en, log_fn, trace_data_pointer)			\
+	_EVENT_SUBSCRIBERS_DEFINE(ename);							\
+	const struct event_type _CONCAT(__event_type_, ename) __used				\
+	__attribute__((__section__("event_types"))) = {						\
+		.name				= STRINGIFY(ename),				\
+		.subs_start	= {								\
+			[_SUBS_PRIO_FIRST]	= _EVENT_SUBSCRIBERS_START			\
+						(ename, _SUBS_PRIO_ID(_SUBS_PRIO_FIRST)),	\
+			[_SUBS_PRIO_NORMAL]	= _EVENT_SUBSCRIBERS_START			\
+						(ename, _SUBS_PRIO_ID(_SUBS_PRIO_NORMAL)),	\
```

Comment:
general comment: would be good to run clang-format, I see multiple inconsistencies all around

Reply:
Looking at this snipped I think this is an old implementation before the rework. Please check the one in NCS for the newest code.
It might be that many things seen here are already gone.